corehq/apps/data_dictionary/tests/test_view.py

```diff
@@ -134,19 +152,52 @@ def test_update_description(self):
     def test_allowed_values_all_valid(self):
         prop = self._get_property()
         prop_payload = self._get_allowed_values_payload(prop)
-        response = self.client.post(self.url, {"properties": json.dumps([prop_payload])})
+        response = self.client.post(self.url, {"groups": '[]', "properties": json.dumps([prop_payload])})
         self.assertEqual(response.status_code, 200)
         self._assert_allowed_values(prop, prop_payload)
 
     def test_allowed_values_with_invalid_one(self):
         prop = self._get_property()
         prop_payload = self._get_allowed_values_payload(prop, all_valid=False)
-        response = self.client.post(self.url, {"properties": json.dumps([prop_payload])})
+        response = self.client.post(self.url, {"groups": '[]', "properties": json.dumps([prop_payload])})
         self.assertEqual(response.status_code, 400)
         response_data = json.loads(response.content)
         self.assertTrue(response_data["messages"][0].startswith("Unable to save valid values longer than"))
         self._assert_allowed_values(prop, prop_payload)
 
+    def test_update_with_group_name(self):
+        prop = self._get_property()
+        self.assertEqual(prop.group, '')
+        self.assertIsNone(prop.group_obj)
+        post_data = {
+            "groups": '[{"caseType": "caseType", "name": "group", "description": ""}]',
+            "properties": '[{"caseType": "caseType", "name": "property", "group": "group"}]'
+        }
+        response = self.client.post(self.url, post_data)
+        self.assertEqual(response.status_code, 200)
+        prop = self._get_property()
+        self.assertEqual(prop.group, 'group')
+        self.assertIsNotNone(prop.group_obj)
+
+    def test_update_with_no_group_name(self):
+        post_data = {
+            "groups": '[{"caseType": "caseType", "name": "group", "description": ""}]',
+            "properties": '[{"caseType": "caseType", "name": "property", "group": "group"}]'
+        }
+        response = self.client.post(self.url, post_data)
```

Comment:
nit: will it be better to update the record directly, instead of sending a request to do the update to setup for the test?

Reply:
Fixed. [9b2e739](https://github.com/dimagi/commcare-hq/pull/32878/commits/9b2e739faaf9ebdb55caaf6c9d3735822469dbcf)